### PR TITLE
docs: Adds vision document

### PR DESCRIPTION
**Pull request by SHession**
## What does this change?

This PR adds an initial version of our ProseMirror-Elements vision document. It provides a clear set of challenges, ambition and guiding principals which can be used to help orient the direction of the project.

## How can we measure success?

When key decisions need to be made, we can refer to this document to see how the different options compare to our guiding principals. It should also provide a clear definition of the problem we are aiming to solve and why.

### CODE REVIEW

File: docs/vision.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
+# Challenges
+
+1. Nesting arbitrary content within ProseMirror 
+2. Seamlessly Integrating nested ProseMirror rich-text fields with the parent instance
```
Comment (paperboyo) [inline docs/vision.md:4]:
```suggestion
2. Seamlessly integrating nested ProseMirror rich-text fields with the parent instance
```

File: docs/vision.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+
+## Easy to implement
+
+Adding the plugin to a new or existing project should be seamless and quickly show results. Once users have set the plugin up, they should be able to easily develop and add new elements.
+
+#### KPIs
+
+- Possible to implement into a ProseMirror instance quickly
+- Creating new elements can be
```
Comment (paperboyo) [inline docs/vision.md:33]: can be… great?

Comment (SHession) [reply docs/vision.md:33]: Looks like I was distracted mid-

File: docs/vision.md [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+#### KPIs
+
+- Possible to implement into a ProseMirror instance quickly
+- Creating new elements can be 
+- Support different requirements
+
+## Agnostic
+
+The plugin should have no opinion about the elements that use it. Any complexity necessary to implement an element should be possible without increasing the surface area of the API.
```
Comment (paperboyo) [inline docs/vision.md:38]: > Any complexity necessary (…) should be possible
No complexity should be possible! 😜

**Review by jonathonherbert — APPROVED** [review]
Reviewed as a mob w/ @dskamiotis, @SHession and @rhystmills 👍

**Pull request merged**